src/Microsoft.Health.Fhir.Shared.Core/Features/Search/Expressions/Parsers/ExpressionParser.cs

```diff
@@ -65,16 +65,43 @@ public Expression Parse(string resourceType, string key, string value)
             return ParseImpl(resourceType, key.AsSpan(), value);
         }
 
+        public IncludeExpression ParseInclude(string resourceType, string includeValue)
+        {
+            var valueSpan = includeValue.AsSpan();
+            if (!TrySplit(SearchSplitChar, ref valueSpan, out ReadOnlySpan<char> originalType))
+            {
+                throw new InvalidSearchOperationException("Type must be specified");
+            }
+
+            SearchParameterInfo refSearchParameter;
+            bool wildCard = false;
+            ReadOnlySpan<char> targetType;
+
+            if (valueSpan.ToString().Equals("*", StringComparison.InvariantCultureIgnoreCase))
+            {
+                refSearchParameter = null;
+                wildCard = true;
+                targetType = string.Empty.AsSpan();
+            }
+            else
+            {
+                TrySplit(SearchSplitChar, ref valueSpan, out targetType);
```

Comment:
the target resource type is optional, so if that's not provided, would this be null?

Reply:
I wasn't originally allowing for that, and my ordering was wrong. I've sense updated such that the targetType is null if it's not supplied and the search will search across all types for the given parameter.